signature/cose/cose.go

```diff
@@ -15,34 +26,524 @@ func init() {
 	}
 }
 
+// Protected Headers
+// https://github.com/notaryproject/notaryproject/blob/cose-envelope/signature-envelope-cose.md
+const (
+	headerKeyExpiry                       = "io.cncf.notary.expiry"
+	headerKeySigningTime                  = "io.cncf.notary.signingTime"
+	headerKeyAuthenticSigningTime         = "io.cncf.notary.authenticSigningTime"
+	headerKeySigningScheme                = "io.cncf.notary.signingScheme"
+	headerKeyVerificationPlugin           = "io.cncf.notary.verificationPlugin"
+	headerKeyVerificationPluginMinVersion = "io.cncf.notary.verificationPluginMinVersion"
+	headerKeyCrit                         = "crit"
+	headerKeyAlg                          = "alg"
+	headerKeyCty                          = "cty"
+)
+
+// Unprotected Headers
+// https://github.com/notaryproject/notaryproject/blob/cose-envelope/signature-envelope-cose.md
+const (
+	headerKeyTimeStampSignature = "io.cncf.notary.timestampSignature"
+	headerKeySigningAgent       = "io.cncf.notary.signingAgent"
+)
+
+// Map of signature.Algorithm to cose.Algorithm
+var signatureAlgCOSEAlgMap = map[signature.Algorithm]cose.Algorithm{
+	signature.AlgorithmPS256: cose.AlgorithmPS256,
+	signature.AlgorithmPS384: cose.AlgorithmES384,
+	signature.AlgorithmPS512: cose.AlgorithmPS512,
+	signature.AlgorithmES256: cose.AlgorithmES256,
+	signature.AlgorithmES384: cose.AlgorithmES384,
+	signature.AlgorithmES512: cose.AlgorithmES512,
+}
+
+var coseAlgSignatureAlgMap = reverseMapCOSE(signatureAlgCOSEAlgMap)
+
+func reverseMapCOSE(m map[signature.Algorithm]cose.Algorithm) map[cose.Algorithm]signature.Algorithm {
+	n := make(map[cose.Algorithm]signature.Algorithm, len(m))
+	for k, v := range m {
+		n[v] = k
+	}
+	return n
+}
+
 type envelope struct {
+	coseEnvelope *cose.Sign1Message
+}
+
+type pluginSigner struct {
+	signer signature.Signer
+}
```

Comment:
We can just call it a signer instead of pluginSigner.

```suggestion
type signer struct {
	base signature.Signer
}
```